test/1.4/misc/test_dmlc_g.dml

```diff
@@ -0,0 +1,44 @@
+/*
+  © 2022 Intel Corporation
+  SPDX-License-Identifier: MPL-2.0
+*/
+// test-dmlc-g.dml - A device containing DML items to test dmlc's -g flag
+
+dml 1.4;
+
+device test;
+
+header %{
+typedef struct sample_interface {
+    void (*simple_method)(conf_object_t *obj, int arg);
+} sample_interface_t;
+%}
+
+extern typedef struct {
+    void (*simple_method)(conf_object_t *obj, int arg);
+} sample_interface_t;
+
+param bank_traitref = cast(reg[1], bank);
+
+session double d = 3.14;
+
+attribute int_attr is int64_attr;
+
+port p_irq {
+    implement sample {}
+}
+
+bank reg[i < 2] {
+    group x {}
+    register r size 4 @ 0x0000 {
+        field EN @ [0];
+    }
+}
+
+connect pic {
+    interface sample;
+}
+
+event ev_cycles is simple_cycle_event {
+    method event() { }
+}
```

Comment:
this test could be more covering of types, I think. In particular Im curious to see what happens if we output debug info for endian integers.

Reply:
Interesting thought, I don't actually know how those are handled. t102 doesn't even test it.